automation/prepare-and-start.sh

```diff
@@ -37,8 +37,8 @@ trap 'cleanup fail' SIGINT SIGTERM
 
 # Create the normal user to be used for bitbake (barys)
 echo "[INFO] Creating and setting builder user $BUILDER_UID:$BUILDER_GID."
-groupadd docker
 groupadd -g $BUILDER_GID builder
```

Comment:
You could specify here "-f" along with "-g" and it will not depend on the order on how you create these groups. From the manpage of groupadd:

> -f, --force
> 
>            This option causes the command to simply exit with success status if the specified group already exists. When used with -g, and the specified GID already exists, another (unique) GID is chosen
>            (i.e.  -g is turned off).
> 

Then, when you user add, manpage of useradd says:

>        -g, --gid GROUP
>            The group name or number of the user's initial login group. The group name must exist. A group number must refer to an already existing group.
> 

So you can simply use the group name previously created.